frontend/src/routes.js

```diff
@@ -7,20 +7,39 @@ import Home from './pages/Home';
 import Challenges from './pages/Challenges';
 import Detail from './pages/Detail';
 import Devs from './pages/Devs';
+import Login from './pages/Login';
+import Submit from './pages/Submit';
+import UserChallenges from './pages/UserChallenges';
+import Dashboard from './pages/Dashboard';
+
+import Header from './components/Header';
+import Footer from './components/Footer';
+
+import { Container } from './styles/GlobalStyles';
 
 function Routes() {
     return (
         <BrowserRouter>
             <ToastContainer />
-            <Switch>
-                <Route path="/" exact component={Home} />
-                <Route path="/challenges" component={Challenges} />
-                <Route path="/challenges/frontend" component={Challenges} />
-                <Route path="/challenges/backend" component={Challenges} />
-                <Route path="/challenges/mobile" component={Challenges} />
-                <Route path="/detail/:id" component={Detail} />
-                <Route path="/devs" component={Devs} />
-            </Switch>
+            <Container>
+                <Switch>
+                    <Route path="/" exact component={Home} />
+                    <Route path="/challenges" component={Challenges} />
+                    <Route path="/challenges/frontend" component={Challenges} />
+                    <Route path="/challenges/backend" component={Challenges} />
+                    <Route path="/challenges/mobile" component={Challenges} />
```

Comment:
Ao invés de colocar a filtragem nos parâmetros da rota, coloque no query.

Existe um problema ao tentar trocar a rota manualmente.
Por exemplo, `/challenges/mobile/` de `/challenges/backend` vai listar todos do mobile normalmente, mas listar todos desafios assim que a rota for alterada.

Corrigir no back-end.